alacritty/src/window_context.rs

```diff
@@ -327,7 +327,7 @@ impl WindowContext {
             return;
         }
 
-        if self.dirty {
+        if self.dirty || terminal.has_damage() {
```

Comment:
The right condition would never be true, when the left is false.

In general all of your damage related changes must be discarded, since they are slowing down perf when actually rendering a frame. I'd suggest to not touch damage the way it's right now, hence it's optimal in terms of perf the way it's right now.


In general I'd suggest to focus on dirty bit.